newIDE/app/src/ObjectEditor/Editors/SpriteEditor/SpritesList.js

```diff
@@ -260,77 +262,67 @@ export default class SpritesList extends Component<Props, void> {
-          const sprite = new gd.Sprite();
-          sprite.setImageName(resource.name);
-          // Restore collision masks and points
-          if (
-            resource.originalIndex !== undefined &&
-            resource.originalIndex !== null
-          ) {
-            const originalSprite = direction.getSprite(resource.originalIndex);
-            copySpritePoints(originalSprite, sprite);
-            copySpritePolygons(originalSprite, sprite);
-          } else {
-            if (allDirectionSpritesHaveSamePoints) {
-              copySpritePoints(direction.getSprite(0), sprite);
-            }
-            if (allDirectionSpritesHaveSameCollisionMasks) {
-              copySpritePolygons(direction.getSprite(0), sprite);
-            }
-          }
-          newDirection.addSprite(sprite);
-          sprite.delete();
-        });
+    if (!editResult) return;
+    const { resources, newMetadata, newName } = editResult;
 
-        // set metadata if there is such on the direction
-        if (resources[0].metadata) {
-          newDirection.setMetadata(JSON.stringify(resources[0].metadata));
+    const newDirection = new gd.Direction();
+    newDirection.setTimeBetweenFrames(direction.getTimeBetweenFrames());
+    newDirection.setLoop(direction.isLooping());
+    resources.forEach(resource => {
+      const sprite = new gd.Sprite();
+      sprite.setImageName(resource.name);
+      // Restore collision masks and points
+      if (
+        resource.originalIndex !== undefined &&
+        resource.originalIndex !== null
+      ) {
+        const originalSprite = direction.getSprite(resource.originalIndex);
+        copySpritePoints(originalSprite, sprite);
+        copySpritePolygons(originalSprite, sprite);
+      } else {
+        if (allDirectionSpritesHaveSamePoints) {
+          copySpritePoints(direction.getSprite(0), sprite);
         }
-
-        // Burst the ResourcesLoader cache to force images to be reloaded (and not cached by the browser).
-        resourcesLoader.burstUrlsCacheForResources(project, resourceNames);
-        onReplaceByDirection(newDirection);
-        // Set optional animation name if the user hasn't done so
-        if (resources[0].newAnimationName) {
-          onChangeName(resources[0].newAnimationName);
+        if (allDirectionSpritesHaveSameCollisionMasks) {
+          copySpritePolygons(direction.getSprite(0), sprite);
         }
-        newDirection.delete();
-      },
+      }
+      newDirection.addSprite(sprite);
+      sprite.delete();
     });
+
+    // Set metadata on the direction to allow editing again in the future.
+    if (newMetadata) {
+      newDirection.setMetadata(JSON.stringify(newMetadata));
+    }
+
+    // Burst the ResourcesLoader cache to force images to be reloaded (and not cached by the browser).
+    resourcesLoader.burstUrlsCacheForResources(project, resourceNames);
+    onReplaceByDirection(newDirection);
+    // Set optional animation name if the user hasn't done so

```

Comment:
the comment seems off, should it be "if a new name is provided, update it"?

Reply:
Updated